src/bosh_azure_cpi/lib/cloud/azure/restapi/azure_client.rb

```diff
@@ -342,12 +349,28 @@ def create_virtual_machine(resource_group_name, vm_params, network_interfaces, a
 
       vm['zones'] = [vm_params[:zone]] unless vm_params[:zone].nil?
 
-      os_disk = {
-        'name' => vm_params[:os_disk][:disk_name],
-        'createOption' => 'FromImage',
-        'caching' => vm_params[:os_disk][:disk_caching]
-      }
-      os_disk['diskSizeGB'] = vm_params[:os_disk][:disk_size] unless vm_params[:os_disk][:disk_size].nil?
+      os_disk = {}
+      unless vm_params[:os_disk].nil?
+        os_disk = {
+          'name' => vm_params[:os_disk][:disk_name],
+          'createOption' => 'FromImage',
+          'caching' => vm_params[:os_disk][:disk_caching]
+        }
+        os_disk['diskSizeGB'] = vm_params[:os_disk][:disk_size] unless vm_params[:os_disk][:disk_size].nil?
+      end
+
+      unless vm_params[:ephemeral_os_disk].nil?
+        os_disk = {
+          'diffDiskSettings' => {
+            'option' => 'Local',
+            'placement' => vm_params[:ephemeral_os_disk][:disk_placement]
+          },
+          'caching' => 'ReadOnly',
```

Comment:
- Maybe we could move this `caching` definition to the `ephemeral_os_disk` function like this is done for the normal ephemeral disks. 
- Is  'caching' => 'ReadOnly' ok for an ephemeral disk where we need also to write?

Reply:
The "caching" option has nothing to do with "read / write " access. 
It's only about optimization / disk iops.
You can only select "read-only" caching for the ephemeral os disk.

Overview about azure disk performance and what kind of impact the disk "caching" has:
https://learn.microsoft.com/en-us/azure/virtual-machines/disks-performance